### PR TITLE
feat: extract NEAR login component from SDK

### PR DESCRIPTION
**Pull request by frdomovic**
#feat: extract NEAR login component from SDK

## Summary:
Extract login component with all necessary helper functions, types and API functions in node client.

#Fixes: [I570](https://github.com/calimero-network/core/issues/570)

#Tests:

https://github.com/user-attachments/assets/4e386a52-66dd-4255-bdba-ce9efcc88953

### CODE REVIEW

**Comment by frdomovic** [comment]
Both login and add root key extracted and implemented with reusable components and custom hooks. 


https://github.com/user-attachments/assets/17aa9934-0902-4710-87c9-99155ec698e4

File: src/main.tsx [hunk 1]
```diff
@@ -3,13 +3,17 @@ import ReactDOM from 'react-dom/client';
...
+import { getNearEnvironment } from './utils/node';
 
 const root = ReactDOM.createRoot(
   document.getElementById('root') as HTMLElement,
 );
 
 root.render(
   <React.StrictMode>
-    <App />
+    <WalletSelectorContextProvider network={getNearEnvironment()}>
```
Comment (frdomovic) [inline src/main.tsx:15]: Wrapper around route + use same component

File: src/components/common/LoginIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+export default function LoginIcon() {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" x="0" y="0" height="30" width="30">
```
Comment (frdomovic) [inline src/components/common/LoginIcon.tsx:5]: add to assets

File: src/components/common/Loading.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+export default function Loading() {
+  const css = `
+  .loader {
```
Comment (frdomovic) [inline src/components/common/Loading.tsx:5]: styled component

File: src/api/dataSource/NodeDataSource.ts [hunk 4]
```diff
@@ -339,4 +420,34 @@ export class NodeDataSource implements NodeApi {
...
+      },
+    );
+  }
+  async requestChallenge(): ApiResponse<NodeChallenge> {
+    return await this.client.post<NodeChallenge>(
+      `${getAppEndpointKey()}/admin-api/request-challenge`,
+      {},
+    );
+  }
+  async addRootKey(rootKeyRequest: NearRequest): ApiResponse<RootKeyResponse> {
```
Comment (frdomovic) [inline src/api/dataSource/NodeDataSource.ts:437]: NearRequest will be updated to LoginRequest naming as this function is used by Metamask aswell

File: src/components/common/Loading.tsx
Comment (alenmestrov) [inline src/components/common/Loading.tsx:1]: Do we need class "loader" here?

Comment (frdomovic) [reply src/components/common/Loading.tsx:1]: updated

**Review by alenmestrov — APPROVED** [review]

File: src/hooks/useNear.ts [hunk 5]
```diff
@@ -106,3 +138,311 @@ export function useRPC() {
...
+    appName,
+    setErrorMessage,
+  }: HandleSignMessageProps) {
+    try {
+      const challengeResponseData: ResponseData<NodeChallenge> = await apiClient
+        .node()
+        .requestChallenge();
+
+      if (challengeResponseData.error) {
+        console.log('requestChallenge api error', challengeResponseData.error);
```
Comment (alenmestrov) [inline src/hooks/useNear.ts:330]: ovdje postaviti error.log i mozda ispisati poruku korisniku

File: src/hooks/useNear.ts [hunk 6]
```diff
@@ -106,3 +138,311 @@ export function useRPC() {
...
+
+  const verifyMessageBrowserWallet = useCallback(
+    async (isLogin: boolean, setErrorMessage: (message: string) => void) => {
+      const urlParams = new URLSearchParams(window.location.hash.substring(1));
+      const accId = urlParams.get('accountId') as string;
+      const publicKey = urlParams.get('publicKey') as string;
+      const signature = urlParams.get('signature') as string;
+
+      if (!accId && !publicKey && !signature) {
+        console.error('Missing params in url.');
```
Comment (alenmestrov) [inline src/hooks/useNear.ts:229]: error.log i errorMessage

File: src/pages/Near.tsx [hunk 7]
```diff
@@ -1,26 +1,97 @@
...
-          nearTitleColor={'white'}
-        />
-      </WalletSelectorContextProvider>
+      <NearWallet
+        isLogin={isLogin}
+        navigateBack={() =>
+          isLogin ? navigate('/auth') : navigate('/identity/root-key')
+        }
+        account={account}
+        accounts={accounts}
```
Comment (alenmestrov) [inline src/pages/Near.tsx:76]: Combine acocuntId and account

File: src/api/dataSource/NodeDataSource.ts [hunk 8]
```diff
@@ -339,4 +420,34 @@ export class NodeDataSource implements NodeApi {
       return { error: { code: 500, message: 'Failed to join context.' } };
```
[the remark this answers is not in the document]
Comment (frdomovic) [reply src/api/dataSource/NodeDataSource.ts:420]: Fixed [here](https://github.com/calimero-network/admin-dashboard/pull/48).
Additionally this whole file needs to be updated with translations but this is out of scope for this task.

File: src/pages/Near.tsx (hunk 7 above)
Comment (frdomovic) [reply src/pages/Near.tsx:76]: Updated

**Pull request merged**